src/pcapi/core/users/repository.py

```diff
@@ -47,20 +48,23 @@ def get_user_with_credentials(identifier: str, password: str) -> User:
 
 
 def get_user_with_valid_token(
-    token_value: str, token_types: list[models.TokenType], delete_token: bool = False
+    token_value: str, token_types: list[models.TokenType], use_token: bool = True
 ) -> Optional[User]:
-    token = models.Token.query.filter(
-        models.Token.value == token_value, models.Token.type.in_(token_types)
-    ).one_or_none()
+    token = (
+        models.Token.query.filter(models.Token.value == token_value, models.Token.type.in_(token_types))
+        .filter_by(isUsed=False)
```

Comment:
Est-ce qu'on peut faire ceci pour garder de la cohérence ?
```suggestion
        models.Token.query.filter(models.Token.value == token_value, models.Token.type.in_(token_types), models.Token.isUsed==False)
```